### DISCUSSION TITLE
How to  update multiple fields using a dictionary

### DISCUSSION THREAD

**Discussion by soorireddy** [sqlalchemy/sqlalchemy#5877] [Usage Questions] [answered]
**Describe your question**

**following is working**
async def update_by_id(self, id: int, record: Dict):
        updateString ="first_name='update_firstname'"
        async with create_async_session() as session:
            query = update(self.model).where(self.model.id == id).values(first_name='update_firstname')
**Not working**
       query = update(self.model).where(self.model.id == id).values(**updateString**)
 besides this would like to update multiple fields.
       query = update(self.model).where(self.model.id == id).values(record)
assume record is {first_name:"update" , last_name:"updateLN", ......}

**Complete stack trace, if applicable**

**Versions**
 - OS:
 - Python:
 - SQLAlchemy: 1.4.Beta
 - Database: MySQL
 - DBAPI:

**Additional context**

**Useful links**
- [Get Support](https://www.sqlalchemy.org/support.html) on the website
- The [documentation](https://docs.sqlalchemy.org/en/latest/) website
- The [UsageRecipes](https://github.com/sqlalchemy/sqlalchemy/wiki/UsageRecipes) wiki
- [Stack Overflow](https://stackoverflow.com/questions/tagged/sqlalchemy) tag
- SQLAlchemy [Google group](http://groups.google.com/group/sqlalchemy)
- [Gitter](https://gitter.im/sqlalchemy/community) chat

**Have a nice day!**

**Comment by CaselIT** [comment, marked as answer]
Hi,

There is no way of using the update string, since SQLAlchemy would not know how to parse it.
You can use the forms
- `query = update(self.model).where(self.model.id == id).values(first_name='update_firstname', last_name='other')`
- `query = update(self.model).where(self.model.id == id).values(record)`
- `query = update(self.model).where(self.model.id == id).values(**record)`

also assuming that record is as you suggested.

See also the [documentation for update](https://docs.sqlalchemy.org/en/14/core/dml.html#sqlalchemy.sql.expression.Update.values)
